File: ANGEN/SampleFramework.Controls/trunk/Samples/Notification Category/Cycler/VB/Cycler/MainForm.vb
Imports Genetibase.Shared.Controls
Imports System.Diagnostics

Public Class MainForm
    Private Sub UpdateProgressBarValue(ByVal progressBar As NuGenProgressBar)
        Debug.Assert(Not progressBar Is Nothing, "progressBar != null")

        If progressBar.Value = progressBar.Maximum Then
            progressBar.Value = progressBar.Minimum
        End If

        progressBar.Value += 1
    End Sub

    Private Sub _goButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles _goButton.Click
        _timer.Enabled = Not _timer.Enabled

        If _timer.Enabled Then
            _goButton.Text = "&Stop"
        Else
            _goButton.Text = "&Start"
        End If
    End Sub

    Private Sub _horzTrackBar_ValueChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles _horzTrackBar.ValueChanged
        _timer.Interval = 500 / _horzTrackBar.Value
    End Sub

    Private Sub _scrollBar_ValueChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles _scrollBar.ValueChanged
        _vertProgressBar.Value = _scrollBar.Maximum - _scrollBar.Value
    End Sub

    Private Sub _timer_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles _timer.Tick
        Me.UpdateProgressBarValue(_horzProgressBar)
        Me.UpdateProgressBarValue(_horzProgressBar2)
    End Sub

    Public Sub New()
        Me.InitializeComponent()
    End Sub
End Class

File: ANGEN/SampleFramework.Controls/SmoothControls - User's Guide/Samples/Notification Category/Cycler/CS/Cycler/MainForm.cs
using Genetibase.Shared.Controls;

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;

namespace Cycler
{
	public partial class MainForm : Form
	{
		private void UpdateProgressBarValue(NuGenProgressBar progressBar)
		{
			Debug.Assert(progressBar != null, "progressBar != null");

			if (progressBar.Value == progressBar.Maximum)
			{
				progressBar.Value = progressBar.Minimum;
			}

			progressBar.Value++;
		}

		private void _goButton_Click(object sender, EventArgs e)
		{
			_timer.Enabled = !_timer.Enabled;

			if (_timer.Enabled)
			{
				_goButton.Text = "&Stop";
			}
			else
			{
				_goButton.Text = "&Start";
			}
		}

		private void _horzTrackBar_ValueChanged(object sender, EventArgs e)
		{
			_timer.Interval = 500 / _horzTrackBar.Value;
		}

		private void _scrollBar_ValueChanged(object sender, EventArgs e)
		{
			_vertProgressBar.Value = _scrollBar.Maximum - _scrollBar.Value;
		}

		private void _timer_Tick(object sender, EventArgs e)
		{
			this.UpdateProgressBarValue(_horzProgressBar);
			this.UpdateProgressBarValue(_horzProgressBar2);
		}

		public MainForm()
		{
			this.InitializeComponent();
		}
	}
}
